cpp/src/cuimage.cpp

```diff
@@ -273,18 +266,18 @@ CuImage::~CuImage()
                 image_data_->container.data = nullptr;
                 break;
             case io::DeviceType::kCUDA:
-                cudaError_t cuda_status;
-                CUDA_TRY(cudaFree(image_data_->container.data));
-                image_data_->container.data = nullptr;
-                if (cuda_status)
+
+                if (image_data_->loader)
                 {
-                    fmt::print(stderr, "[Error] Cannot free memory!");
+                    cudaError_t cuda_status;
+                    CUDA_TRY(cudaFree(image_data_->container.data));
                 }
+                image_data_->container.data = nullptr;
                 break;
             case io::DeviceType::kPinned:
             case io::DeviceType::kCPUShared:
             case io::DeviceType::kCUDAShared:
-                fmt::print(stderr, "Device type {} is not supported!", device_type);
+                fmt::print(stderr, "Device type {} is not supported!\n", device_type);
```

Comment:
JFYI there is an issue about this upstream ( https://github.com/fmtlib/fmt/issues/428 )

Reply:
Oh, that's good to know! Thanks for the pointer!